### ISSUE TITLE
schema-reader Error: Target class [db] does not exist.

### ISSUE THREAD

**Issue by Xety** [VemtoOrg/vemto2-issues#20]
Hey, I reported you my error on Twitter. With the 2.0.4 it's still doesn't work, but I have the error message now.

**Error Message:**
schema-reader Error: Target class [db] does not exist.

**Stack:**
```
schema-reader Error: #0 Z:\Web\Sites\bds\vendor\laravel\framework\src\Illuminate\Container\Container.php(795): Illuminate\Container\Container->build('db')
#1 Z:\Web\Sites\bds\vendor\laravel\framework\src\Illuminate\Foundation\Application.php(957): Illuminate\Container\Container->resolve('db', Array, true)
#2 Z:\Web\Sites\bds\vendor\laravel\framework\src\Illuminate\Container\Container.php(731): Illuminate\Foundation\Application->resolve('db', Array)
#3 Z:\Web\Sites\bds\vendor\laravel\framework\src\Illuminate\Foundation\Application.php(942): Illuminate\Container\Container->make('db', Array)
#4 Z:\Web\Sites\bds\vendor\laravel\framework\src\Illuminate\Container\Container.php(1454): Illuminate\Foundation\Application->make('db')
#5 Z:\Web\Sites\bds\vendor\laravel\framework\src\Illuminate\Database\MigrationServiceProvider.php(64): Illuminate\Container\Container->offsetGet('db')
#6 Z:\Web\Sites\bds\vendor\laravel\framework\src\Illuminate\Container\Container.php(908): Illuminate\Database\MigrationServiceProvider->Illuminate\Database\{closure}(Object(Illuminate\Foundation\Application), Array)
#7 Z:\Web\Sites\bds\vendor\laravel\framework\src\Illuminate\Container\Container.php(795): Illuminate\Container\Container->build(Object(Closure))
#8 Z:\Web\Sites\bds\vendor\laravel\framework\src\Illuminate\Foundation\Application.php(957): Illuminate\Container\Container->resolve('migration.repos...', Array, true)
#9 Z:\Web\Sites\bds\vendor\laravel\framework\src\Illuminate\Container\Container.php(731): Illuminate\Foundation\Application->resolve('migration.repos...', Array)
#10 Z:\Web\Sites\bds\vendor\laravel\framework\src\Illuminate\Foundation\Application.php(942): Illuminate\Container\Container->make('migration.repos...', Array)
#11 Z:\Web\Sites\bds\vendor\laravel\framework\src\Illuminate\Container\Container.php(1454): Illuminate\Foundation\Application->make('migration.repos...')
#12 Z:\Web\Sites\bds\vendor\laravel\framework\src\Illuminate\Database\MigrationServiceProvider.php(79): Illuminate\Container\Container->offsetGet('migration.repos...')
#13 Z:\Web\Sites\bds\vendor\laravel\framework\src\Illuminate\Container\Container.php(908): Illuminate\Database\MigrationServiceProvider->Illuminate\Database\{closure}(Object(Illuminate\Foundation\Application), Array)
#14 Z:\Web\Sites\bds\vendor\laravel\framework\src\Illuminate\Container\Container.php(795): Illuminate\Container\Container->build(Object(Closure))
#15 Z:\Web\Sites\bds\vendor\laravel\framework\src\Illuminate\Foundation\Application.php(957): Illuminate\Container\Container->resolve('migrator', Array, true)
#16 Z:\Web\Sites\bds\vendor\laravel\framework\src\Illuminate\Container\Container.php(731): Illuminate\Foundation\Application->resolve('migrator', Array)
#17 Z:\Web\Sites\bds\vendor\laravel\framework\src\Illuminate\Foundation\Application.php(942): Illuminate\Container\Container->make('migrator', Array)
#18 Z:\Web\Sites\bds\vendor\laravel\framework\src\Illuminate\Container\Container.php(1454): Illuminate\Foundation\Application->make('migrator')
#19 phar://C:/Users/Emeric/AppData/Local/Programs/Vemto/resources/static/VMTTL1/index.php(42): Illuminate\Container\Container->offsetGet('migrator')
#20 phar://C:/Users/Emeric/AppData/Local/Programs/Vemto/resources/static/VMTTL1/common/Vemto.php(107): {closure}()
#21 phar://C:/Users/Emeric/AppData/Local/Programs/Vemto/resources/static/VMTTL1/index.php(18): Vemto::execute('schema-reader', Object(Closure))
#22 C:\Users\Emeric\AppData\Local\Programs\Vemto\resources\static\VMTTL1(14): require('phar://C:/Users...')
#23 {main}
```

---

Vemto 2.0.4
win32 10.0.22631
Locale: fr
App version: 2.0.4

**Comment by TiagoSilvaPereira** [comment]
@Xety is this a new project or an existing one? what Laravel version is this project? When Vemto tries to read the migrations, it first tries to boot the Laravel application. Seems the error is happening during the application boot.

**Comment by Xety** [comment]
> @Xety is this a new project or an existing one? what Laravel version is this project? When Vemto tries to read the migrations, it first tries to boot the Laravel application. Seems the error is happening during the application boot.

It's an existing one.

Laravel v10.48.2

My project can be found here if needed : https://github.com/Xety/bds

**Comment by TiagoSilvaPereira** [comment]
> > @Xety is this a new project or an existing one? what Laravel version is this project? When Vemto tries to read the migrations, it first tries to boot the Laravel application. Seems the error is happening during the application boot.
> 
> It's an existing one.
> 
> Laravel v10.48.2
> 
> My project can be found here if needed : https://github.com/Xety/bds

Oh thanks, this will help a lot!!

**Comment by TiagoSilvaPereira** [comment]
@Xety, I just checked, and the problem occurs due to some migration methods that the schema reader cannot interpret (for example, $this->driver()).

At the moment, I'm creating another schema reader with a different philosophy, which theoretically should work in any project, regardless of how the migrations were created. I will let you know as soon as I have any news about this.

**Comment by TiagoSilvaPereira** [comment]
Fixed on 2.0.7 by using Database Schema Reader mode
